### PR TITLE
Fixes TS Mother gateway crossover

### PR DESCRIPTION
**Pull request by Kyep**
🆑 Kyep
fix: Evil-minded players can no longer use the Gateway's Mother of Terror spider to create an infestation on-station.
/ 🆑

Also refactors some 1 into TRUE and 0 into FALSE.

### CODE REVIEW

**Comment by SamHPurp** [comment]
To confirm this fixes when people kill the mother in the gateway room offstation, and the spiderlings travel through? If so... 👍 x 💯

**Comment by Kyep** [comment]
This prevents:

- Kiting the Mother to right next to the Gateway, then letting her wander across the threshold onto the Cyberiad, die there, and spawn spiderlings there.
- Killing the mother in a small confined space in the gateway room, such that her spiderlings spawn in the mission, but can randomly walk around the room - and walk through the gateway teleporter, only to find themselves on the Cyberiad, and grow up there.

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/mother.dm [hunk 1]
```diff
@@ -33,12 +33,15 @@
 	ventsmash_action.Grant(src)
 
 /mob/living/simple_animal/hostile/poison/terror_spider/mother/death(gibbed)
+	var/always_stillborn = 0
```
Comment (SamHPurp) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/mother.dm]: `0` = `FALSE`

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/mother.dm [hunk 2]
```diff
@@ -33,12 +33,15 @@
...
+	if(spider_awaymission && !is_away_level(z))
+		always_stillborn = 1
```
Comment (SamHPurp) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/mother.dm]: `TRUE`

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/reproduction.dm [hunk 3]
```diff
@@ -16,10 +16,13 @@
 	var/use_vents = 1
 	var/list/enemies = list()
 	var/immediate_ventcrawl = 0
+	var/spider_awaymission = 0
```
Comment (SamHPurp) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/reproduction.dm]: `FALSE`

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/reproduction.dm [hunk 4]
```diff
@@ -16,10 +16,13 @@
...
 
 /obj/structure/spider/spiderling/terror_spiderling/New()
 	..()
 	ts_spiderling_list += src
+	if(is_away_level(z))
+		spider_awaymission = 1
```
Comment (SamHPurp) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/reproduction.dm]: `TRUE`

File: code/modules/mob/living/simple_animal/hostile/terror_spiders/reproduction.dm [hunk 5]
```diff
@@ -85,6 +88,8 @@
 	if(isturf(loc))
 		amount_grown += rand(0,2)
 		if(amount_grown >= 100)
+			if(spider_awaymission && !is_away_level(z))
+				stillborn = 1
```
Comment (SamHPurp) [inline code/modules/mob/living/simple_animal/hostile/terror_spiders/reproduction.dm]: `TRUE`

**Comment by Kyep** [comment]
Changed.

**Comment by LightFire53** [comment]
I would think the RD teleport gun would still be valid. People bringing spiders via the gateway because they are bored is a no go though.

**Pull request merged**